sdk/database/dbplugin/v5/testing/test_helpers.go

```diff
@@ -22,19 +22,47 @@ func getRequestTimeout(t *testing.T) time.Duration {
 	return dur
 }
 
-func AssertInitialize(t *testing.T, db dbplugin.Database, req dbplugin.InitializeRequest) dbplugin.InitializeResponse {
+// AssertInitializeCircleCiTest help to diagnose CircleCI failures within AssertInitialize for mssql tests failing
+// with "Failed to initialize: error verifying connection ..."
+func AssertInitializeCircleCiTest(t *testing.T, db dbplugin.Database, req dbplugin.InitializeRequest) dbplugin.InitializeResponse {
 	t.Helper()
+	maxAttempts := 5
+	var resp dbplugin.InitializeResponse
+	var err error
+
+	for i := 1; i <= maxAttempts; i++ {
+		resp, err = verifyInitialize(t, db, req)
+		switch {
+		case err != nil:
+			t.Logf("Failed AssertInitialize attempt: %d with error:\n%+v\n", i, err)
+			time.Sleep(1 * time.Second)
+		case i == 1:
+			return resp
+		default:
+			t.Fatalf("AssertInitialize worked the %d time around with a 1 second sleep, but failed originally", i)
```

Comment:
should this be a `t.Fatalf`? that would end the test. would we instead like to log it and return the successful response, or do we want to stop the test if it didn't work the first time?

Reply:
Yes the t.Fatalf is intentional. Nick and I agreed, we still wanted to fail the test to provide visibility when it would occur, with the extra debugging hopefully providing additional insights.